### PR TITLE
0.39 release

### PR DESCRIPTION
**Pull request by natestemen**
## Description

Release notes and such for 0.39 release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
 
 ## Version 0.39.0
-_In development_
+
+<!-- ([Full Changelog](https://github.com/unitaryfund/mitiq/compare/v0.38.0...v0.39.0)) -->
+
+### Highlights
+
+We've made updates to our documentation, beginning with the completion of the first section of the Pauli Twirling user guide, which offers a comprehensive introduction to this feature.
+Additionally, we've added a new tutorial on CDR (Clifford Data Regression) with [Qrack](https://github.com/unitaryfund/qrack/), providing users with an in-depth look at how to integrate these tools effectively.
```
Comment (Misty-W) [inline CHANGELOG.md:10]: Might be nice to mention that Qrack is a near-Clifford simulator, so the power of (mostly) efficient simulation in training can be realized.

**Review by Misty-W — APPROVED** [review]
Looks good @natestemen ! 

Just one non-blocking suggestion on the description of the CDR tutorial.

File: CHANGELOG.md (hunk 1 above)
Comment (natestemen) [reply CHANGELOG.md]: Great suggestion, thanks! Changed in [`fc0a282`](https://github.com/unitaryfund/mitiq/pull/2457/commits/fc0a2825d1d2c82d9d0a3f49df8eedc86b462108).

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,31 @@
 # Changelog
 
 ## Version 0.39.0
-_In development_
+
+<!-- ([Full Changelog](https://github.com/unitaryfund/mitiq/compare/v0.38.0...v0.39.0)) -->
```
Comment (cosenal) [inline CHANGELOG.md:5]: There is actually a problem with this that @jordandsullivan spotted and reported in #2438.

Comment (natestemen) [reply CHANGELOG.md]: I like the solution you posted [here](https://github.com/unitaryfund/mitiq/issues/2438#issuecomment-2272108429). Implemented in https://github.com/unitaryfund/mitiq/pull/2457/commits/c3d514d24828ba7912c682573af5c06472ad39a3.

**Review by cosenal — APPROVED** [review]
🚢

**Pull request merged**